src/utils/line.ts

```diff
@@ -785,3 +776,45 @@ export const formatOptimisticProposal = (
 
   return proposedPosition;
 };
+
+// TODO: Move code from getLines to create state.categories into this function
+export const formatLineCategories = (
+  lines: { [key: string]: SecuredLine },
+  categories: { [key: string]: string[] }
+): { [key: string]: string[] } => {
+  const validLines = _.filter(lines, (line) => line.status === 'active' && line.end * 1000 > Date.now());
```

Comment:
you should do this filtering in the graphql query directly so we only get back valid data and then we dont need this extra code

Reply:
done!